src/object/cli_obj.c

```diff
@@ -1745,11 +1745,15 @@ obj_retry_cb(tse_task_t *task, struct dc_object *obj,
 	     struct obj_auxi_args *obj_auxi, bool pmap_stale,
 	     bool *io_task_reinited)
 {
-	tse_sched_t	 *sched = tse_task2sched(task);
-	tse_task_t	 *pool_task = NULL;
-	uint32_t	  delay;
-	int		  result = task->dt_result;
-	int		  rc;
+	daos_event_t	*ev = dc_task_get_event(task);
+	tse_sched_t	*sched = tse_task2sched(task);
+	tse_task_t	*pool_task = NULL;
+	uint32_t	 delay;
+	int		 result = task->dt_result;
+	int		 rc;
+
+	if (unlikely(ev != NULL && ev->ev_abort))
+		D_GOTO(err, rc = -DER_CANCELED);
```

Comment:
if it is in dead loop in retry of object IO here, then the API task seems should not be finished and should not get to user's daos_event_priv_reset() call. 
In the ticket pasted log is not detailed enough so not very sure if really related with this.

Reply:
Related task is not completed, but the test logic long jumped out of the context, that is related with cmocka logic for handling some signal. I will remake the patch to handle that.